### PR TITLE
test: add JSON5 specs test suite

### PR DESCRIPTION
**Pull request by aqrln**
This commit adds and adapts the official JSON5 parser tests.
`test/todo/json5.js` should be moved to `test/node` as soon as we fix
all the failures.

### CODE REVIEW

**Comment by aqrln** [comment]
@metarhia/jstp-core I am going to go ahead and land this by the end of the day but I'd like to get at least two reviews.

File: test/todo/json5.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const tap = require('tap');
+const jstp = require('../..');
+
+const supportedByUs = {
```
Comment (belochub) [inline test/todo/json5.js]: We should add noctal number tests here, because we decided before that we support them.

**Comment by aqrln** [comment]
@belochub @lundibundi I've added noctal tests.

**Comment by aqrln** [comment]
Landed in e9db526d5cfe0bed4bbd97c9a8df0f3b875b04fc. Thanks for the reviews!

**Pull request closed without merging**